nodes/modifier_make/contour2D.py

```diff
@@ -0,0 +1,492 @@
+            newAngs = sorted(newAngs + netAll)
+
+            for angL in newAngs:
+                listVertX.append(x + Radius*cos(angL))
+                listVertY.append(y + Radius*sin(angL))
+                listVertZ.append(z)
+
+        elif connex == 1:
+
+            beta = (net[1] - net[2])
+            if beta <= 0:
+                beta = (net[1] - net[2] + 2*pi)
+            vAngle += net[0] - net[1] - pi
+            vert = int((degrees(beta)) / theta) + 1
+
+            for i in range(vert):
+                localAngle = (radians(theta*i) + vAngle + 4*pi) % (2*pi)
+                listVertX.append(x + Radius * cos(localAngle))
+                listVertY.append(y + Radius * sin(localAngle))
+                listVertZ.append(z)
+
+            listVertX.append(x + Radius * cos(net[1]))
+            listVertY.append(y + Radius * sin(net[1]))
+            listVertZ.append(z)
+
+        else:
+            for i in range(vert):
+                localAngle = (radians(theta*i) + vAngle + 4*pi) % (2*pi)
+
+                listVertX.append(x + Radius * cos(localAngle))
+                listVertY.append(y + Radius * sin(localAngle))
+                listVertZ.append(z)
+        points = list((x, y, z) for x, y, z in zip(listVertX, listVertY, listVertZ))
+        return points
+
+    def side_edges(self, v, edges, radius, net):
+        Verts = []
+        Edges = []
+        n = 0
+        netOffset = [0 for i in range(len(net))]
+
+        for ed in edges:
+            for s in ed:
+                netOf = netOffset[s]
+                netOffset[s] += 3
+                x1 = v[s][0] + radius[s] * cos(net[s][1 + netOf])
+                y1 = v[s][1] + radius[s] * sin(net[s][1 + netOf])
+                x2 = v[s][0] + radius[s] * cos(net[s][2 + netOf])
+                y2 = v[s][1] + radius[s] * sin(net[s][2 + netOf])
+
+                Verts.append((x1, y1, v[s][2]))
+                Verts.append((x2, y2, v[s][2]))
+
+            Edges.append((n, n+3))
+            Edges.append((n+1, n+2))
+            n += 4
+        return Verts, Edges
+
+    def make_edges(self, Vertices, net, pointsL):
+        listEdg = [(i, i + 1) for i in range(pointsL - 1)]
+        if len(net) != 1:
+            listEdg.append((pointsL-1, 0))
+        return listEdg
+
+    def process(self):
+        inputs, outputs = self.inputs, self.outputs
+
+        # inputs
+        vertsAll = inputs['Verts_in'].sv_get(deepcopy=False, default=[[(0.0, 0.0, 0.0)]])
+
+        radiusAll = inputs['Distance'].sv_get(deepcopy=False, default=[[abs(self.rad_)]])
+        radiusAll = [list(map(lambda x: abs(x), radius)) for radius in radiusAll]
+
+        verticesAll = inputs['Nº Vertices'].sv_get(deepcopy=False, default=[[self.vert_]])
+        verticesAll = [list(map(lambda x: max(2, int(x)), Vertices)) for Vertices in verticesAll]
+
+        edgesAll = inputs['Edges_in'].sv_get(deepcopy=False, default=[[]])
+
+        # outputs
+        if not outputs['Vertices'].is_linked:
```

Comment:
this check can go directly at the top, stops all further processing if no outputs linked,